src/DynamoCoreWpf/UI/Converters.cs

```diff
@@ -1581,20 +1581,36 @@ public object Convert(object value, Type targetType, object parameter, CultureIn
             // if the number of directories deep exceeds threshold
             if (str.Length - str.Replace(@"\", "").Length >= 5)
             {
-                var root = Path.GetPathRoot(str);
+                //directories to go down under the root
+                var threshold = 2;
                 var name = Path.GetFileName(str);
-
-                var dirInfo = new DirectoryInfo(Path.GetDirectoryName(str));
-
-                var collapsed = new[]
+                
+                var currentDirInfo = new DirectoryInfo(Path.GetDirectoryName(str));
+                var root = currentDirInfo.Root;
+                var rootName = root.FullName;
+
+                var collapsed = new List<string>();
+                collapsed.Add(name);
+                
+                var count = 0;
+                while(count < threshold)
```

Comment:
Prefer for loops to while loops. They're more concise, easier to read and harder to make some kinds of mistakes with.

for (int count = 0; count < threshold; count++) {

...
}